ansible_base/authentication/authenticator_plugins/local.py

```diff
@@ -69,5 +87,84 @@ def authenticate(self, request, username=None, password=None, **kwargs):
                     "is_superuser": user.is_superuser,
                 },
             )
-
         return update_user_claims(user, self.database_instance, [])
+
+    def _can_authenticate_from_controller(self, username, password):
+        """
+        Check if a user exists in the AuthenticatorUser table with the local authenticator provider.
+        If the user is valid, update the gateway users credentials with the controller credentials.
+        """
+        try:
+            user = UserModel._default_manager.get_by_natural_key(username)
+        except UserModel.DoesNotExist:
+            logger.warning(f"User '{username}' does not exist in the database.")
+            return False
+
+        if controller_user := self._get_controller_user(username, password):
+            # Validate controller_user is a dictionary
+            if not isinstance(controller_user, dict):
+                logger.warning(f"Unable to authenticate user '{username}' with controller. Invalid result format.")
+                return False
+            # Validate controller_user has a ldap_dn field, if it is not None, then the user is a local user
+            ldap_dn = controller_user.get("ldap_dn")
+            if (ldap_dn is None or ldap_dn != "") and user.password == "$encrypted$":
+                logger.warning(f"User '{username}' is unable to be confirmed as a local user.")
+                return False
+            return True
+        else:
+            return False
+
+    def _get_controller_user(self, username: str, password: str):
+        """
+        Get the user from the controller by making a request to the controller API /me/ endpoint.
+        If the user is not found, return None.
+        If the user is found, return the user.
+        """
+
+        controller_base_domain = get_setting('gateway_proxy_url')
+        if not controller_base_domain:
+            logger.warning("Controller authentication failed, unable to get controller base domain")
+            return None
+        controller_url = urljoin(controller_base_domain, "/api/controller/v2/me/")
+
+        try:
+            response = requests.get(controller_url, auth=HTTPBasicAuth(username, password), timeout=10)
+            response.raise_for_status()
+            user_data = response.json()
+
+            # Check if count exists and equals 1
+            count = user_data.get("count")
+            if count != 1:
+                logger.warning(f"Unable to authenticate user '{username}' with controller.")
+                return None
+
+            # Check if results exists and is a non-empty list
+            results = user_data.get("results")
+            if not results or not isinstance(results, list) or len(results) == 0:
+                logger.warning(f"Unable to authenticate user '{username}' with controller. Invalid or empty results.")
```

Comment:
```suggestion
                logger.info(f"Unable to authenticate user '{username}' with controller. Invalid or empty results.")
```

I'm thinking info log for this one. This one can happen if someone types the wrong password. It's not necessarily a system issue